### PR TITLE
Correct channelIncompatibility for geojson type

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: src/fielddef.ts [hunk 1]
```diff
@@ -576,6 +576,13 @@ const COMPATIBLE = {compatible: true};
 export function channelCompatibility(fieldDef: FieldDef<Field>, channel: Channel): {compatible: boolean; warning?: string;} {
   const type = fieldDef.type;
 
+  if (type === 'geojson' && channel !== 'shape') {
```
Comment (domoritz) [inline src/fielddef.ts]: Why does color not work?

Comment (kanitw) [reply src/fielddef.ts]: Why does it work?  Isn't geojson only for path shape?

Comment (domoritz) [reply src/fielddef.ts]: I thought geojson can have properties that can be encoded with the color of the shape for instance.

Comment (kanitw) [reply src/fielddef.ts]: For those properties, shouldn't the `type` be other typical type (not geojson)?

Comment (domoritz) [reply src/fielddef.ts]: Oh, I see. Yes.

**Pull request merged**